types/src/v0/v0_3/chain_config.rs

```diff
@@ -0,0 +1,175 @@
+
+    /// Account that receives sequencing bids.
+    pub bid_recipient: Option<FeeAccount>,
+}
+
+#[derive(Clone, Debug, Copy, PartialEq, Deserialize, Serialize, Eq, Hash)]
+/// A commitment to a ChainConfig or a full ChainConfig.
+pub struct ResolvableChainConfig {
+    pub(crate) chain_config: Either<ChainConfig, Commitment<ChainConfig>>,
+}
+
+impl Committable for ChainConfig {
+    fn tag() -> String {
+        "CHAIN_CONFIG".to_string()
+    }
+
+    fn commit(&self) -> Commitment<Self> {
+        let comm = committable::RawCommitmentBuilder::new(&Self::tag())
+            .fixed_size_field("chain_id", &self.chain_id.to_fixed_bytes())
+            .u64_field("max_block_size", *self.max_block_size)
+            .fixed_size_field("base_fee", &self.base_fee.to_fixed_bytes())
+            .fixed_size_field("fee_recipient", &self.fee_recipient.to_fixed_bytes());
+        let comm = if let Some(addr) = self.fee_contract {
+            comm.u64_field("fee_contract", 1).fixed_size_bytes(&addr.0)
+        } else {
+            comm.u64_field("fee_contract", 0)
+        };
+
+        // With `ChainConfig` upgrades we want commitments w/out
+        // fields added >= v0_3 to have the same commitment as <= v0_3
+        // commitment. Therefore `None` values are simply ignored.
+        let comm = if let Some(bid_recipient) = self.bid_recipient {
+            comm.fixed_size_field("bid_recipient", &bid_recipient.to_fixed_bytes())
+        } else {
+            comm
+        };
+
+        comm.finalize()
+    }
+}
+
+impl ResolvableChainConfig {
+    pub fn commit(&self) -> Commitment<ChainConfig> {
+        match self.chain_config {
+            Either::Left(config) => config.commit(),
+            Either::Right(commitment) => commitment,
+        }
+    }
+    pub fn resolve(self) -> Option<ChainConfig> {
+        match self.chain_config {
+            Either::Left(config) => Some(config),
+            Either::Right(_) => None,
+        }
+    }
+}
+
+impl From<Commitment<ChainConfig>> for ResolvableChainConfig {
+    fn from(value: Commitment<ChainConfig>) -> Self {
+        Self {
+            chain_config: Either::Right(value),
+        }
+    }
+}
+
+impl From<ChainConfig> for ResolvableChainConfig {
+    fn from(value: ChainConfig) -> Self {
+        Self {
+            chain_config: Either::Left(value),
+        }
+    }
+}
+
+impl From<&v0_1::ResolvableChainConfig> for ResolvableChainConfig {
+    fn from(
+        &v0_1::ResolvableChainConfig { chain_config }: &v0_1::ResolvableChainConfig,
+    ) -> ResolvableChainConfig {
+        match chain_config {
+            Either::Left(chain_config) => ResolvableChainConfig {
+                chain_config: Either::Left(ChainConfig::from(chain_config)),
+            },
+            // TODO does this work? is there a better way?
+            Either::Right(c) => ResolvableChainConfig {
+                chain_config: Either::Right(Commitment::from_str(&c.to_string()).unwrap()),
```

Comment:
I also haven't found a better way to handle this. maybe @jbearer would know

Reply:
I think this is ok to merge for now, but we try to figure out if there is a better way before we turn on marketplace.